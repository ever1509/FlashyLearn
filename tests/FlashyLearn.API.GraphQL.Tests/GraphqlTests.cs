using HotChocolate;
using HotChocolate.Execution;
using Infrastructure.GraphQL;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace FlashyLearn.API.GraphQL.Tests;

public class GraphqlTests
{
    [Fact]
    public async Task SchemaChangeTest()
    {
        var schema = await new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .AddFiltering()
            .BuildSchemaAsync();
        
        schema.ToString().MatchSnapshot();
    }

    [Fact]
    public async Task FetchAllCategories()
    {
        var result = await new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .AddFiltering()
            .ExecuteRequestAsync(@"  allCategories {
                                                categoryID,
                                                name,
                                                userID
                                            }");
        result.ToJson().MatchSnapshot();
    }
}