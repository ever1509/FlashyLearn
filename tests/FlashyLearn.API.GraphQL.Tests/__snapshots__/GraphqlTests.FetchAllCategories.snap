{
  "errors": [
    {
      "message": "Unexpected token: Name.",
      "locations": [
        {
          "line": 1,
          "column": 3
        }
      ],
      "extensions": {
        "code": "HC0014"
      }
    }
  ]
}
